update_nightly_list.py

```diff
@@ -0,0 +1,79 @@
+#! /usr/bin/env python3
+# -*- coding: utf-8 -*-
+# vim:fenc=utf-8
+#
+# Copyright © 2018 Martine Lenders <m.lenders@fu-berlin.de>
+#
+# Distributed under terms of the MIT license.
+
+import argparse
+import datetime
+import json
+import os
+import re
+import time
+
+FILENAME = "nightlies"
+
+
+def find_commit_build(builds, commit):
+    for build in builds:
+        if build["commit"] == commit:
+            return build
+    return None
+
+
+def main(args):
+    nightlies_list_file = os.path.join(args.repodir, FILENAME)
+    try:
+        # parse existing list file
+        with open(nightlies_list_file) as f:
+            nightlies = json.load(f)
+    except:
+        # there is no list file at the moment or a different error happened
+        # just try to create a new one
+        nightlies = {}
+    for branch in args.branch:
+        branch_dir = os.path.join(args.repodir, branch)
+        if branch not in nightlies:
+            # if branch is not yet in the nightlies list, create an instance
+            nightlies[branch] = []
+        # get hash of latest commit from symlink
+        latest_commit = os.path.realpath(os.path.join(branch_dir, "latest"))\
+            .split(os.path.sep)[-1]
+        # check if it was already build in a previous build
+        build = find_commit_build(nightlies[branch], latest_commit)
+        if build is None:
+            build = {"commit": latest_commit}
```

Comment:
I actually like to also have the commit summary and author here, but I wasn't sure where (if anywhere) to access the git repository.

Reply:
Same goes for the `output_url`, but I know where I can get that from (I just forgot about it)

Reply:
Murdock does have that information, maybe we can make it export those.

Reply:
Where are they?

Reply:
Murdock exports quite some info via environment: 
https://github.com/kaspar030/murdock:/blob/d2655fe7f7204374ec7e65810d89e3e3aba8c0e6/murdock.py#L214

The PR author github name should already be available as ```CI_PULL_USER```

Here would be the place to get more fields from the PR json:
https://github.com/kaspar030/murdock/blob/d2655fe7f7204374ec7e65810d89e3e3aba8c0e6/murdock.py#L274